parcels/field.py

```diff
@@ -566,11 +572,19 @@ def search_indices_curvilinear(self, x, y, z, xi, yi, ti=-1, time=-1, search2D=F
                          [1, -1, 1, -1]])
         maxIterSearch = 1e6
         it = 0
+        if not grid.zonal_periodic or grid.mesh == 'flat':
```

Comment:
So we do not support `flat` grids that are periodic? I would argue that there are many cases where flat grids can be periodic (e.g. zonal channels etc)

Reply:
We do support periodic flat grids. (the test in test_fieldset is an example of such a grid). It is just that the out-of-bound tests here shouldn't be used in spherical, zonal-periodic grids.